scripts/actions/send-and-update-translation-queue.js

```diff
@@ -0,0 +1,200 @@
+const path = require('path');
+const FormData = require('form-data');
+const fetch = require('node-fetch');
+
+const loadFromDB = require('./utils/load-from-db');
+const saveToDB = require('./utils/save-to-db');
+const serializeMDX = require('./serialize-mdx');
+const { vendorRequest, getAccessToken } = require('./utils/vendor-request');
+
+/**
+ * @typedef {Object} Page
+ * @property {string} file The filepath for the page (from the project root).
+ * @property {string} html The HTML serialized content for the page.
+ */
+
+// NOTE: the vendor requires the locales in a different format
+// We should consider this into the Gatsby config for each locale.
+const LOCALE_IDS = {
+  jp: 'ja-JP',
+};
+
+const PROJECT_ID = process.env.TRANSLATION_VENDOR_PROJECT;
+
+/**
+ * Take a list of filepaths (grouped by locale) and fetches the HTML content.
+ * NOTE: async inside of loops is hard.
+ * @param {Object<string, string[]>} locales The queue of slugs to be translated.
+ * @returns {Promise<Object<string, Page[]>>}
+ */
+const getContent = async (locales) =>
+  Object.entries(locales).reduce(async (accP, [locale, slugs]) => {
+    const acc = await accP;
+    return {
+      ...acc,
+      [locale]: await Promise.all(
+        slugs.map(async (slug) => {
+          const mdx = fs.readFileSync(path.join(process.cwd(), slug));
+          const html = await serializeMDX(mdx);
+          return {
+            file: slug,
+            html: html ? html : false,
+          };
+        })
+      ),
+    };
+  }, {});
+
+/**
+ * @param {string} locale The locale that this file should be translated to.
+ * @param {string} batchUid The batch that is expecting this file.
+ * @param {string} accessToken
+ * @returns {(page: Page) => Promise<string>} A function that turns a page into a request.
+ */
+const uploadFile = (locale, batchUid, accessToken) => async (page) => {
+  const filepath = path.join(process.cwd(), 'to-translate.html');
+  fs.writeFileSync(filepath, page.html, 'utf8');
+
+  const form = new FormData();
+  form.append('fileType', 'html');
+  form.append('localeIdsToAuthorize[]', LOCALE_IDS[locale]);
+  form.append('fileUri', page.file);
+  form.append('file', fs.createReadStream(filepath));
+
+  const url = new URL(
+    `/job-batches-api/v2/projects/${PROJECT_ID}/batches/${batchUid}/file`,
+    process.env.TRANSLATION_VENDOR_API_URL
+  );
+
+  const options = {
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+    },
+    body: form,
+  };
+
+  try {
+    const resp = await fetch(url.href, options);
+    const { response } = await resp.json();
+    const { code } = response;
+
+    if (code !== 'ACCEPTED') {
+      console.dir(response, { depth: null });
+      throw new Error(code);
+    }
+
+    console.log(`[*] Successfully uploaded ${page.file}.`);
+    return code;
+  } catch (error) {
+    console.error(`[!] Unable to upload ${page.file}.`);
+    console.error(error);
+    process.exit(1);
```

Comment:
I wonder if we should gather all of the jobs that failed to upload and instead do this at the end of the script. By doing this here, this will halt immediately, which likely would kill any in-progress uploads. It might be best to try and get as many uploaded as we can.

Also by exiting here, it looks like you aren't able to record which of these completed successfully, which means any time we re-run this script, all of the jobs previously successful uploads will happen again. We may want to [record](https://github.com/newrelic/docs-website/pull/581/files#diff-3275c521907ff8a8e4237b2b80241b357a478e451ec141a3c21300fdfcbb7df0R186) which ones succeeded as they finish so that we avoid sending them if something fails in the middle.

Reply:
I'm returning the `code` for all uploads and using it for a log:
```
[*] Successfully uploaded 10 / 12 files
```
This should allow for partial success with the file uploads.